src/main/java/org/sunbird/extension/util/TransformJsonUtil.java

```diff
@@ -0,0 +1,698 @@
+package org.sunbird.extension.util;
+
+import com.google.common.collect.HashBiMap;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.responsecode.ResponseCode;
+
+/**
+ * Util to transform the structure of a map into another, based on the configuration
+ *
+ * @author Jaikumar Soundara Rajan
+ * @version 1.0
+ */
+public class TransformJsonUtil {
+
+  private static final Pattern listTypePattern = Pattern.compile("<(.*)>");
+
+  /**
+   * All fields available in userInputMap that are configured in config as well, will be transformed
+   * according to the configuration and returned as a map
+   *
+   * @param fieldsConfig Config
```

Comment:
Complete Javadoc for params and return type.

Reply:
Review comment Incorporated